odo/backends/sql.py

```diff
@@ -172,7 +172,9 @@ def discover_typeengine(typ):
     if isinstance(typ, (sa.NUMERIC, sa.DECIMAL)):
         return datashape.Decimal(precision=typ.precision, scale=typ.scale)
     if isinstance(typ, (sa.String, sa.Unicode)):
-        return datashape.String(typ.length, typ.collation)
+        return datashape.String(typ.length, None)
+    if isinstance(typ, sa.dialects.mssql.BIT):
+        return datashape.int_
```

Comment:
I don't know mssql at all but the docs imply that this should be `datashape.bool_`